docs/plugins.md

```diff
@@ -0,0 +1,215 @@
+# Authoring Plugins
+
+A published plugin must be its own installable `npm` module, following the
+plugin [naming convention]. You may also create [one-off plugins] for an app.
+
+A plugin's main export has the following shape:
+
+```js
+export default {
+  name: 'awesome',
+  dependencies: [/* list of plugin names */],
+  hooks: {
+    hookA(gasket) {
+
+    },
+    async hookB(gasket) {
+
+    },
+    hookC: {
+      timing: {},
+      handler(gasket) {
+
+      }
+    }
+  }
+};
+```
+
+The optional `dependencies` array lists other plugins that must be installed as
+a prerequisite. The `hooks` map has a key for every event the plugin handles,
+with the values being either the function itself or an object specifying
+`timing` options and the `handler`.
+
+## Naming convention
+
+Plugins and presets should adhere to the project-type prefixed naming convention. This formatting allows user plugins to be referenced with short names and will help avoid collisions. This convention also mimics those by other projects such a `@babel` and `@oclif`
```

Comment:
The naming convention is not a hard requirement anymore since we import plugins. Can you rework this as a "recommended naming convention" with some examples?